src/ServiceControl/Monitoring/HeartbeatMonitoringHostedService.cs

```diff
@@ -10,7 +10,7 @@
 
     class HeartbeatMonitoringHostedService : IHostedService
     {
-        public HeartbeatMonitoringHostedService(EndpointInstanceMonitoring monitor, MonitoringDataStore persistence, IAsyncTimer scheduler, Settings settings)
+        public HeartbeatMonitoringHostedService(EndpointInstanceMonitoring monitor, IMonitoringDataStore persistence, IAsyncTimer scheduler, Settings settings)
```

Comment:
To be able to switch between different implementations, do we need to create (and use here) a Factory instead?